ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorFragment.kt

```diff
@@ -55,11 +55,15 @@ class SurveySelectorFragment : AbstractFragment(), BackPressListener {
         showProgressDialog()
       }
       UiState.SurveyActivated -> {
-        dismissProgressDialog()
+        if (!viewModel.surveyActivationInProgress.value) {
```

Comment:
Why would `SurveyActivated` state be set while the survey activation is still in progress? Is this check really just masking the root cause?

Reply:
Not masking the root cause

User selects a survey -> ActivatingSurvey State -> Show progress dialog -> **SurveyListAvailable State** (**this state is getting emitted again, due to re-arrangment of the survey activation list, emitting SurveyListAvailable state and the dialog is getting dismissed before onSuccess**) -> onSuccess -> SurveyActivated State -> navigate to home

Reply:
Ok, so it sounds like there's a circular state dependency.. perhaps the list should simplfy stop updating once a survey is tapped?

Reply:
Ok, but perhaps the UI state should pertain to the initial load and not reactive updates? The root cause seems to be the circle dependency on the state flow and the user tap. Ideally we would break that loop.